sympy/utilities/codegen.py

```diff
@@ -528,16 +531,17 @@ def _get_symbol(self, s):
                 raise ValueError("Failed to print %s" % str(s))
         return expr_str.strip()
 
-    def __init__(self, project="project"):
+    def __init__(self, project="project", cse=False):
         """Initialize a code generator.
 
         Derived classes will offer more options that affect the generated
         code.
 
         """
         self.project = project
+        self.cse = cse
 
-    def routine(self, name, expr, argument_sequence, global_vars):
+    def routine(self, name, expr, argument_sequence=None, global_vars=None):
```

Comment:
btw I can easily be talked out of allowing the argument_sequence to be none as you will end up with something random. that might sufficient if you are developing code you will dump somewhere else, though.